### PR TITLE
Allow setting a prefix for temp directory

### PR DESCRIPTION
**Pull request by Richienb**
Signed-off-by: Richie Bendall <richiebendall@gmail.com>

cc @sindresorhus

### CODE REVIEW

**Comment by sindresorhus** [comment]
What's the use-case for this? What problem are you trying to solve?

**Comment by Richienb** [comment]
@sindresorhus 

Allow developers to easily find the cache directories created by a particular application for faster debugging.

**Comment by sindresorhus** [comment]
Debugging of what? Can you describe how you would use this for debugging? Just a name prefix will not help much as there will probably be more than one directory.

**Comment by Richienb** [comment]
@sindresorhus Having a prefix would allow us to know which application created the directory.

**Comment by sindresorhus** [comment]
Yes, but what do you do with that information?

**Comment by Richienb** [comment]
@sindresorhus 

> Yes, but what do you do with that information?

In my use case, my script automatically downloads and extracts a ffmpeg binary to a temporary path so, with a directory prefix, I can easily find the directory that it downloaded it to.

File: index.d.ts [hunk 1]
```diff
@@ -19,6 +19,15 @@ declare namespace tempy {
 			readonly name?: string;
 		}
 	>;
+
+	type DirectoryOptions = {
+		/**
+		_You usually won't need this option. Specify it only when actually needed._
```
Comment (sindresorhus) [inline index.d.ts:25]: This is missing from the readme.

File: index.d.ts [hunk 2]
```diff
@@ -19,6 +19,15 @@ declare namespace tempy {
...
+
+		Directory prefix.
```
Comment (sindresorhus) [inline index.d.ts:27]: The doc comment should also elaborate on the use-cases for this option.

File: index.js [hunk 3]
```diff
@@ -4,7 +4,7 @@ const path = require('path');
 const uniqueString = require('unique-string');
 const tempDir = require('temp-dir');
 
-const getPath = () => path.join(tempDir, uniqueString());
+const getPath = prefix => path.join(tempDir, (prefix || '') + uniqueString());
```
Comment (sindresorhus) [inline index.js:7]:
```suggestion
const getPath = (prefix = '') => path.join(tempDir, prefix + uniqueString());
```

File: index.js [hunk 4]
```diff
@@ -23,8 +23,13 @@ module.exports.file = options => {
 	return getPath() + '.' + options.extension.replace(/^\./, '');
 };
 
-module.exports.directory = () => {
-	const directory = getPath();
+module.exports.directory = options => {
```
Comment (sindresorhus) [inline index.js:26]:
```suggestion
module.exports.directory = ({prefix = ''} = {}) => {
```

File: test.js [hunk 5]
```diff
@@ -13,6 +13,7 @@ test('.file()', t => {
 
 test('.directory()', t => {
 	t.true(tempy.directory().includes(tmpdir()));
+	t.true(tempy.directory({prefix: 'name_'}).includes(tmpdir()));
```
Comment (sindresorhus) [inline test.js:16]: This should test that the prefix was included.

File: index.d.ts (hunk 1 above)
Comment (Richienb) [reply index.d.ts:25]: Fixed!

File: index.d.ts (hunk 2 above)
Comment (Richienb) [reply index.d.ts:27]: Fixed!

File: test.js (hunk 5 above)
Comment (Richienb) [reply test.js]: Fixed!

File: index.js (hunk 4 above)
Comment (Richienb) [reply index.js:26]: Changed!

File: index.js (hunk 3 above)
Comment (Richienb) [reply index.js]: Applied!

**Comment by Richienb** [comment]
@sindresorhus Ready for re-review!

**Comment by sindresorhus** [comment]
Travis is failing

**Pull request merged**